solution-2.test.js

```diff
@@ -0,0 +1,39 @@
+const debounce = require('./solution-2');
+
+test('debounce should delay function execution', async () => {
+  const mockFn = jest.fn();
+  const delayedFn = debounce(mockFn, 100);
+
+  delayedFn();
+  delayedFn();
+
+  await new Promise(resolve => setTimeout(resolve, 150));
```

Comment:
what is the need of this promise? similarly for other promises in the test file?

Reply:
Considering the async behavior of the code, I considered using promises was the best way to go about it. But I'll use faketimers which is much more effective.